components/Offer/Summary.tsx

```diff
@@ -12,14 +12,19 @@ const Summary: FC<
     }
     isSingle: boolean
     price: BigNumber
-    quantity?: number | string
+    quantity?: number | string | BigNumber
```

Comment:
I got a 500 errors when going to the bid page of an erc1155:
https://nft-test-polygon-mumbai-git-feature-number-inpu-19b559-liteflow.vercel.app/tokens/80001-0xe3fe92dfec700e7a168f4b074ee7daca71a11397-42728329798185476104569684045014434977602555793693133601813101861374056018692/bid

It's working well on the bid page of a erc721:
https://nft-test-polygon-mumbai-git-feature-number-inpu-19b559-liteflow.vercel.app/tokens/80001-0x7c68c3c59ceb245733a2fdeb47f5f7d6dbcc65b3-42728329798185476104569684045014434977602555793693133601797249465506198024448/bid

I think the reason is this file line 58 is displaying `quantity` without converting it to a string / number first. This doesn't work if it's a BigNumber.
Should be something like:
```
{quantity?.toString()}
```

Reply:
Also some simplification in this file can be done if you change the type of quantity to:
```suggestion
    quantity: BigNumber
```
This only requires to pass `quantityBN` in `Checkout.tsx` as well.
This will remove the try catch needed line 22-27.
The whole totalPrice can be simply:
```tsx
const totalPrice = useMemo(() => price.mul(quantity), [quantity, price])
```
And `if (!totalPrice) return BigNumber.from(0)` can be removed from `totalFees`'s useMemo